File: Travel/Travel.Models/TripTicket/TripTicket.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Travel.Models.Account;
using Travel.Models.OrganizedTrip;

namespace Travel.Models.TripTicket
{
    public class TripTicket
    {
        public long Id { get; set; }
        public string PassengerId { get; set; }
        public UserResponse Passenger { get; set; }
        public double Price { get; set; }
        public long TripId { get; set; }
        public OrganizedTrip.OrganizedTrip Trip { get; set; }
        public long AgencyId { get; set; }
        public Agency.Agency Agency { get; set; }
        public int NumberOfPassengers { get; set; }
    }
}

File: Travel/UI/travel_admin/lib/model/trip_ticket/trip_ticket.dart
import 'dart:ffi';

import 'package:json_annotation/json_annotation.dart';
import 'package:travel_admin/model/account/account.dart';
import 'package:travel_admin/model/agency/agency.dart';
import 'package:travel_admin/model/organized_trip/organized_trip.dart';

part 'trip_ticket.g.dart';

@JsonSerializable()
class TripTicketModel {
  int? id;
  int? numberOfPassengers;
  int? agencyId;
  AgencyModel? agency;
  String? passengerId;
  AccountModel? passenger;
  double? price;
  int? tripId;
  OrganizedTripModel? trip;
  TripTicketModel(this.id, this.agency, this.agencyId, this.numberOfPassengers,
      this.passenger, this.passengerId, this.price, this.trip, this.tripId);

  /// A necessary factory constructor for creating a new User instance
  /// from a map. Pass the map to the generated `_$UserFromJson()` constructor.
  /// The constructor is named after the source class, in this case, User.
  factory TripTicketModel.fromJson(Map<String, dynamic> json) =>
      _$TripTicketModelFromJson(json);

  /// `toJson` is the convention for a class to declare support for serialization
  /// to JSON. The implementation simply calls the private, generated
  /// helper method `_$UserToJson`.
  Map<String, dynamic> toJson() => _$TripTicketModelToJson(this);
}
